repair/repair.cc

```diff
@@ -1950,7 +1950,33 @@ future<> repair_service::do_rebuild_replace_with_repair(locator::token_metadata_
                 rs.get_metrics().replace_total_ranges = nr_ranges_total;
             }).get();
         }
+        std::unordered_map<sstring, std::unordered_set<gms::inet_address>> source_dc_racks;
+        std::unordered_set<gms::inet_address> source_dc_live_nodes;
+        std::unordered_map<sstring, size_t> lost_nodes_per_rack;
+        const auto& topology = tmptr->get_topology();
+        if (!source_dc.empty()) {
+            try {
+                source_dc_racks = topology.get_datacenter_racks().at(source_dc);
+            } catch (...) {
+                rlogger.warn("{}: Could not find racks for source_dc={}: {}", op, source_dc, std::current_exception());
+                source_dc = "";
+            }
+        }
         rlogger.info("{}: started with keyspaces={}, source_dc={}, nr_ranges_total={}, ignore_nodes={}", op, ks_erms | boost::adaptors::map_keys, source_dc, nr_ranges_total, ignore_nodes);
+        if (!source_dc.empty()) {
+            topology.for_each_node([&] (const locator::node* node) {
+                if (node->dc_rack().dc != source_dc) {
+                    return;
+                }
+                const auto& rack = node->dc_rack().rack;
+                // This node is counted as "lost" since it replaces a dead node
```

Comment:
or this node is an empty node which runs rebuild.

Reply:
yes, will note that